mars/services/subtask/core.py

```diff
@@ -106,13 +106,28 @@ class SubtaskResult(Serializable):
     bands: List[BandType] = ListField("band", FieldTypes.tuple)
     error = AnyField("error", default=None)
     traceback = AnyField("traceback", default=None)
+    # The following is the execution information of the sub task
+    execution_start_time: float = Float64Field("execution_start_time")
+    execution_end_time: float = Float64Field("execution_end_time")
 
     def merge_bands(self, result: Optional["SubtaskResult"]):
         if result and result.bands:
             bands = self.bands or []
             self.bands = sorted(set(bands + result.bands))
+            self._update_result(result)
         return self
 
+    def _update_result(self, result: Optional["SubtaskResult"]):
+        """Update result to save the latest successful execution result."""
+        if (
+            self.status != SubtaskStatus.succeeded
+            and result.status == SubtaskStatus.succeeded
+        ):
```

Comment:
`execution_end_time` shall be set even if `result.ststus` is errored or cancelled.

Reply:
Agree with you. I'll fix it.

Reply:
I just remove `_update_result` method and put them in the `if` block for simplicity and clarity.